cli/commands/analyze-loading-optimization.mjs

```diff
@@ -0,0 +1,527 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import puppeteer, { Browser } from 'puppeteer';
+import round from 'lodash-es/round.js';
+
+/* eslint-disable jsdoc/valid-types */
+/** @typedef {import("web-vitals").LCPMetricWithAttribution} LCPMetricWithAttribution */
+/* eslint-enable jsdoc/valid-types */
+
+/**
+ * Internal dependencies
+ */
+import {
+	isValidTableFormat,
+	log,
+	table,
+	OUTPUT_FORMAT_TABLE,
+} from '../lib/cli/logger.mjs';
+
+/**
+ * @typedef {Object} Device
+ * @property {string}  userAgent User-Agent.
+ * @property {number}  width     Width.
+ * @property {number}  height    Height.
+ * @property {boolean} isMobile  Is mobile.
+ */
+
+/**
+ * @type {Object<string, Device>}
+ */
+const devices = {
+	mobile: {
+		width: 360,
+		height: 800,
+		userAgent:
+			'Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.5938.140 Mobile Safari/537.36',
+		isMobile: true,
+	},
+	desktop: {
+		width: 1920,
+		height: 1080,
+		userAgent:
+			'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36',
+		isMobile: false,
+	},
+};
+
+/**
+ * Command-line arguments.
+ *
+ * @type {Object}
+ */
+export const options = [
+	{
+		argname: '-u, --url <url>',
+		description: 'A URL to check',
+	},
+	{
+		argname: '-o, --output <output>',
+		description: 'Output format: csv, csv-oneline, table, or json', // TODO: Add ability to output CSV as single row for sake of putting in spreadsheet.
+		defaults: OUTPUT_FORMAT_TABLE,
+	},
+];
+
+/**
+ * @typedef  {Object} Params
+ * @property {string} url    - See above.
+ * @property {string} output - See above.
+ */
+
+/**
+ * @param {Object} opt
+ * @param {string} opt.url
+ * @param {string} opt.output
+ * @return {Params} Parameters.
+ */
+function getParamsFromOptions( opt ) {
+	const params = {
+		url: opt.url,
+		output: opt.output,
+	};
+
+	if (
+		! isValidTableFormat( params.output ) &&
+		! [ 'csv-oneline', 'json' ].includes( params.output )
+	) {
+		throw new Error(
+			`Invalid output ${ opt.output }. The output format provided via the --output (-o) argument must be either "table", "csv", "csv-oneline", or "json".`
+		);
+	}
+
+	if ( ! params.url ) {
+		throw new Error(
+			'You need to provide a URL to analyze via the --url (-u) argument.'
+		);
+	}
+
+	return params;
+}
+
+/**
+ * @param {Object} opt
+ */
+export async function handler( opt ) {
+	const params = getParamsFromOptions( opt );
+	const browser = await puppeteer.launch( {
+		headless: 'new',
+		// TODO: Command is not working when opening in non-headless mode. The LCP never fires.
+		// headless: false, devtools: true
```

Comment:
Why does that work when using `benchmark-web-vitals` though? I think it may be good to figure that out so that we can use headless.

Reply:
Actually, the situation is the reverse. It _does_ work with `headless`. It just doesn't work when headful. So it works the same as with `benchmark-web-vitals`.